book/website/reproducible-research/reviewing/reviewing-workflow.md

```diff
@@ -2,44 +2,65 @@
+In the GitHub system, the review is begun directly from and often accessed through the [pull request page](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request).
+The review step occurs between the points where the coder believes their contribution is complete and where that contribution is merged into the trunk code for the project, and so it is intimately associated with a single pull request.
 
-A reviewer should check these things, but defects on these fronts should be by occasional oversight, rather than systematic.
+Within the Github environment, projects can be configured to *require* a review before a given pull request can be merged.
+Even if this option hasn't been selected, it's still possible (and indeed best practice) to manually request a review on a pending pull request.
 
 <a name="create_discuss_change"></a>
-### Create and discuss the review; make the changes
+### Create and discuss the review
 
-At this point, the review process can begin. In Github, the reviewer can provide both general comments as well as line-by-line comments.
+At this point, the review process can begin. In Github, the reviewer can provide both general comments as well as line-by-line comments, see [GitHub code review](https://github.com/features/code-review).
 Each comment becomes its own comment thread, permitting back-and-forth discussion about each issue as required.
 This interaction should allow consensus to be reached on every comment.
-In most cases, the reviewer has final say if a consensus cannot be found.
+
+Once the review is complete, you can discuss any comments necessary. Then you make the changes, and record the changes made against appropriate comments. Also, you check that the reviewer knows you believe you have fully addressed the review.
+
+Once you believe changes are complete, the reviewer check that they do indeed address all of the initial comments. As needed, the reviewer engages constructively with you if they disagree on certain points in order to come to a consensus. In most cases, the reviewer has a final say if a consensus cannot be found.
 
 Once post-review changes have been made to the code, make final updates the comments as needed to complete a history of what has been done and the reasoning behind it.
 
+#### Communicating results through GitHub
+
+In Github, comments should be added in the `Files changed` section, so they can be attached to a particular line of code, see [GitHub reviewing changes in pullrequests](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/reviewing-changes-in-pull-requests). Make many small comments this way, rather than a big ball of text with everything in it, so that different issues can be kept separate. Where relevant, refer to existing Issues and documentation.
```

Comment:
Just an funny comment, I'm just feeling like in Inception movie, using the pull request revision mechanism to review the text which explains just the very same mechanism.

Reply:
(or like Bastian reading Neverending Story book inside Neverending Story book, when he realizes it is telling about himself)

Reply:
I was also thinking about this when updating this chapter :-)